test/plugin/scenarios/webflux-scenario/webflux-projectB-scenario/src/main/java/org/apache/skywalking/apm/testcase/sc/webflux/projectB/controller/TestWebclientController.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.skywalking.apm.testcase.sc.webflux.projectB.controller;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.ExchangeStrategies;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+@RestController
+public class TestWebclientController {
+
+    @Value("${projectB.host:localhost:18080}")
+    private String hostBAddress;
+
+    @RequestMapping("/testcase/webclient/ping")
+    public String ping() {
+        return "ok";
+    }
+
+    @RequestMapping("/testcase/webclient/test1")
+    public String test1() {
+        Mono<String> resp = WebClient.builder().exchangeStrategies(ExchangeStrategies.builder()
+                .build())
+                .build()
+                .get().uri("http://" + hostBAddress + "/testcase/webclient/ping")
```

Comment:
This is the place I have doubts. You have `webclient/test1` entry span, and linked to the previous, so, `visit` method works. And in here, you call `webclient/ping`, 
You can see here
https://github.com/apache/skywalking/blob/4a08b24c6ad2630dc51af0c1fc10bee6e3de74f2/test/plugin/scenarios/webflux-scenario/config/expectedData.yaml#L238-L255
This is a separated segment and span, no parent span and no ref. Meaning, the way you instrumented can not access the context here, which indicates there is another thread executing the codes. 

Is this more clear for you?

Reply:
Thanks. It's very clear. I replace test with this: 
```
@RequestMapping("/testcase/webclient/test1")
    public String test1() {
        CloseableHttpClient httpclient = HttpClients.createDefault();
        try {
            HttpGet httpget = new HttpGet("http://" + hostBAddress + "/testcase/webclient/ping");
            ResponseHandler<String> responseHandler = new ResponseHandler<String>() {
                public String handleResponse(
                        org.apache.http.HttpResponse response) throws ClientProtocolException, IOException {
                    HttpEntity entity = response.getEntity();
                    return entity != null ? EntityUtils.toString(entity) : null;
                }

            };
            return httpclient.execute(httpget, responseHandler);
        } finally {
            httpclient.close();
        }
        return "ok";
    }
```
It's still not link. It's weird. Would you mind testing it locally?

Reply:
I'm testing it locally

Reply:
https://github.com/apache/skywalking/blob/e356da4b0e264a7617b3356febe6d6e25361f20a/apm-sniffer/apm-sdk-plugin/spring-plugins/spring-webflux-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/DispatcherHandlerHandleMethodInterceptor.java#L70
https://github.com/apache/skywalking/blob/e356da4b0e264a7617b3356febe6d6e25361f20a/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java#L179-L184

Hi @wu-sheng @dengliming `AbstractTracerContext` in `DefaultExchangeFunctionInterceptor# beforeMethod` was removed

Reply:
It's still not link because Controller and WebClient calling run in different threads and the WebClient plugin doesn't link the trace context. I suggest that you should save the `ContextSnapshot` when create WebClient Object and `continued` when WebClient send request.

I also found another problem that the trace miss span about WebClient plugin,  In my case, the trace will miss the database span.
```java
 @RequestMapping("/testcase/webclient/test1")
    public String test1() {
        Mono<String> resp = WebClient.builder().exchangeStrategies(ExchangeStrategies.builder()
                .build())
                .build()
                .get().uri("http://" + hostBAddress + "/testcase/webclient/ping")
                .retrieve().bodyToMono(String.class);
        resp.subscribe(new Consumer<String>() {
            @Override
            public void accept(final String s) {
                // read data from database 
            }
        });
        return "ok";
    }
```

Reply:
Hi~ @ascrutae Thanks for reply. I agree with your second point, Now I haven't found a good way to pass the context in Reactor Mono Flow, This is a tricky one.
But the first point, I have also tested WebFlux locally, and the link will also be missing, which has nothing to do with the use of WebClient. This should be an existing bug

Reply:
That Controller and WebClient calling running in different threads are completely random and this is the characteristic of Reactor Mono Flow.

Reply:
@dengliming You have the context in the `#visit`, as the ref exists. I am not sure what do you mean about the bug. One way or another, that is totally different thing. Let's focus on the client issue now.

Reply:
@dengliming Any update?

Reply:
Not yet. Sorry for the late reply. I tried ways to enhance WebFilter and WebClient ExchangeFilterFunction, But the client span still not link. That's probably because I'm missing something or we should also add reactor-core plugin passing the context on between Mono subscriber. I would really appreciate it if you have some ideas about how to do that or you can just modify it on this pr
Any way, We can just keep this PR open for now and see if anyone shares some of their solutions or close it for now if that make no sense. Thanks.

Reply:
@ascrutae Could you take a deep look at this?

Reply:
ok, Got it